### DISCUSSION TITLE
ModbusRTU Server response delay

### DISCUSSION THREAD

**Discussion by dima-sz** [eModbus/eModbus#322] [Q&A] [answered]
Hello, can I reduce the response latency of the Modbus RTU server?

![BAUDRATE 9600](https://github.com/eModbus/eModbus/assets/71876221/ba00a925-8bcd-4a16-b744-70686902bdfb)

![BAUDRATE 19200](https://github.com/eModbus/eModbus/assets/71876221/d303f9b1-bc2f-4379-bdd8-16c0c56429a2)

![BAUDRATE 38400](https://github.com/eModbus/eModbus/assets/71876221/daffc5da-84e4-4d7b-8799-a18f4fd22507)

![BAUDRATE 57600](https://github.com/eModbus/eModbus/assets/71876221/10c8b1d6-e2d4-4386-b738-abc028b7f803)

**Comment by Miq1** [comment, marked as answer]
Unfortunately: no. There is a minimum quiet time between messages mandated by the Modbus RTU standard amounting to 1750 microseconds. This needs to be maintained for standard-conforming servers to detect message ends.

If you know what you are doing, you could modify that for your local installation only. This is done [here in RTUutils.cpp](https://github.com/eModbus/eModbus/blob/dae597ace2cb9b9a3a4034b9cba09c5bdeb00fca/src/RTUutils.cpp#L111), but be warned you will be missing messages if you get to or even below the time for 3.5 characters to be transmitted. And of course you can not rely on any Modbus RTU device in the field to be responsive any more...

**Comment by dima-sz** [comment]
Thank you very much, great result!

![BAUDRATE 57600(2)](https://github.com/eModbus/eModbus/assets/71876221/385a2034-eed6-4fea-b240-3fe403aa77c3)
